full/Dockerfile

```diff
@@ -203,49 +201,65 @@ ENV PATH=$PATH:/home/gitpod/.nvm/versions/node/v${NODE_VERSION}/bin
+
+ENV GP_PYTHON_SHORT_VERSION=3.7 \
+    PATH=$HOME/.pyenv/bin:$HOME/.pyenv/shims:/home/gitpod/.local/bin:$PATH
+
+RUN export GP_PYTHON_VERSION="3.7.7" \
+    && sudo apt-get update \
+    && sudo apt-get install -yq libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev \
+    && curl -fsSL https://www.python.org/ftp/python/${GP_PYTHON_VERSION}/Python-${GP_PYTHON_VERSION}.tar.xz -o Python-${GP_PYTHON_VERSION}.tar.xz \
+    && tar xsf Python-${GP_PYTHON_VERSION}.tar.xz \
+    && cd Python-${GP_PYTHON_VERSION} \
+    && ./configure \
+        --enable-optimizations \
+        --enable-shared \
+        --with-lto \
+        --enable-ipv6 \
+        --enable-loadable-sqlite-extensions \
+        --with-ensurepip=install \
+    && sudo make -j 8 \
+    && sudo make install \
+    && sudo ldconfig \
+    && cd .. \
+    && sudo rm -rf Python-${GP_PYTHON_VERSION} Python-${GP_PYTHON_VERSION}.tar.xz \
+    && curl -fsSL https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
+    && sudo python3 get-pip.py \
+    && rm get-pip.py \
+    && sudo python3 -m pip install --upgrade \
+        setuptools wheel virtualenv pipenv pylint rope flake8 \
+        mypy autopep8 pep8 pylama pydocstyle bandit notebook \
+        python-language-server[all]==0.31.9 twine \
+    && sudo mkdir /home/gitpod/.cache \
+    && sudo chown gitpod:gitpod /home/gitpod/.cache \
+    && sudo chown gitpod:gitpod /home/gitpod/.local/lib/python${GP_PYTHON_SHORT_VERSION}/site-packages \
+    && curl -fsSL https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash \
     && { echo; \
         echo 'eval "$(pyenv init -)"'; \
         echo 'eval "$(pyenv virtualenv-init -)"'; } >> /home/gitpod/.bashrc.d/60-python \
-    && pyenv update \
-    && pyenv install 2.7.17 \
-    && pyenv install 3.7.7 \
-    && pyenv global 2.7.17 3.7.7 \
-    && python2 -m pip install --upgrade pip \
-    && python3 -m pip install --upgrade pip \
-    && python3 -m pip install --upgrade \
-        setuptools wheel virtualenv pipenv pylint rope flake8 \
-        mypy autopep8 pep8 pylama pydocstyle bandit notebook \
-        python-language-server[all]==0.25.0 twine \
-    && sudo rm -rf /tmp/*
-# Gitpod will automatically add user site under `/workspace` to persist your packages.
-# ENV PYTHONUSERBASE=/workspace/.pip-modules \
-#    PIP_USER=yes
+    && sudo rm -rf /tmp/* /var/lib/apt/lists/*
 
 ### Ruby ###
 LABEL dazzle/layer=lang-ruby
 LABEL dazzle/test=tests/lang-ruby.yaml
 USER gitpod
 RUN curl -sSL https://rvm.io/mpapis.asc | gpg --import - \
-    && curl -sSL https://rvm.io/pkuczynski.asc | gpg --import - \
+    && curl -fsSL https://rvm.io/pkuczynski.asc | gpg --import - \
     && curl -fsSL https://get.rvm.io | bash -s stable \
     && bash -lc " \
         rvm requirements \
-        && rvm install 2.5 \
         && rvm install 2.6 \
         && rvm use 2.6 --default \
```

Comment:
In 2019, `2.5` still seemed like the most used Ruby version: https://www.jetbrains.com/lp/devecosystem-2019/ruby/

I think it's valuable to have two Ruby versions: the one that's used by the most developers, and the latest one that should be used.

Unless we definitely know that Ruby 2.5 projects work without any change with the latest Ruby too. (E.g., we've removed Java 8 from our images, even though it's still used by most developers, because we assume that Java projects will also work well with latest stable Java.)

Reply:
Fair enough